File: src/network/packet/Packet.php
<?php

namespace pocketcloud\cloud\network\packet;

use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\util\PacketData;

interface Packet {

    public function encode(PacketData $packetData): void;

    public function encodePayload(PacketData $packetData): void;

    public function decode(PacketData $packetData): void;

    public function decodePayload(PacketData $packetData): void;

    public function handle(ServerClient $client): void;

    public function getName(): string;

    public function isEncoded(): bool;

    public function getSentTimestamp(): ?int;
}

File: cloud-core/src/test/java/packet/Packet.java
package packet;

import packet.util.PacketData;

/**
 * Base interface for all packets in the network system.
 */
public interface Packet {

    void encode(PacketData packetData);

    void encodePayload(PacketData packetData);

    void decode(PacketData packetData);

    void decodePayload(PacketData packetData);

    void handle();

    String getName();

    boolean isEncoded();

    Long getSentTimestamp();
}
